staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager_test.go

```diff
@@ -732,3 +734,52 @@ func TestApplySuccessWithNoManagedFields(t *testing.T) {
 		t.Fatalf("failed to apply object: %v", err)
 	}
 }
+
+// Run an update and apply, and make sure that nothing has changed.
+func TestNoOpChanges(t *testing.T) {
+	f := NewTestFieldManager(schema.FromAPIVersionAndKind("v1", "Pod"))
+
+	obj := &unstructured.Unstructured{Object: map[string]interface{}{}}
+	if err := yaml.Unmarshal([]byte(`{
+		"apiVersion": "v1",
+		"kind": "Pod",
+		"metadata": {
+			"labels": {
+				"a": "b"
+			},
+		}
+	}`), &obj.Object); err != nil {
+		t.Fatalf("error decoding YAML: %v", err)
+	}
+	if err := f.Apply(obj, "fieldmanager_test_apply", false); err != nil {
+		t.Fatalf("failed to apply object: %v", err)
+	}
+	before := f.liveObj.DeepCopyObject()
+	// Wait to make sure the timestamp is different
+	time.Sleep(time.Second)
```

Comment:
ಠ_ಠ

Reply:
We're planning on inserting a Clock to mock with a fake clock and have more tests here. Not much time right now :-/